### PR TITLE
`Development`: Fix flaky postgres test

### PR DESCRIPTION
**Pull request by Strohgelaender**
#### General
- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

### Motivation and Context
The test `testResetCategories` is flaky on postgess.

### Description
The test compared a Set against a List, leading to test failures by expecting the Set to be ordered. The test also used a bit of an hack to compare `StaticCodeAnalysisCategory` objects with `StaticCodeAnalysisDefaultCategory` which I replaced by a explicit mapping. Lastly, I converted `StaticCodeAnalysisDefaultCategory` into a record.

### Steps for Testing
code review

#### Code Review
- [x] Code Review 1
- [x] Code Review 2

## Summary by CodeRabbit

- **Refactor**
	- Simplified and improved the code structure for static code analysis by utilizing Java streams and records for mapping and updating category properties.
- **Bug Fixes**
	- Ensured consistency and accuracy in handling static code analysis categories by adjusting method calls and property access.
- **Tests**
	- Updated test methods to validate the static code analysis functionalities accurately post the recent code enhancements.

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/service/StaticCodeAnalysisService.java [hunk 1]
```diff
@@ -46,16 +46,7 @@ public void createDefaultCategories(ProgrammingExercise programmingExercise) {
         }
 
         // Create new static code analysis using the default configuration as a template
```
Comment (florian-glombik) [inline src/main/java/de/tum/in/www1/artemis/service/StaticCodeAnalysisService.java:48]: I think we could get rid of this commend by renaming `newCategories`, e.g. to `newCategoriesWithDefaultConfiguration` - but I guess this is a matter of taste 🤷‍♂️

**Review by florian-glombik — DISMISSED** [review]
Code changes are looking good to me, using a record seems reasonable and like a better choice.

Thanks for using explicit typing instead of `var` 😄

**Review by pzdr7 — APPROVED** [review]
Code

**Review by laadvo — APPROVED** [review]
Code looks good, thanks for fixing this common cause of flakyness :)

**Pull request merged**